server/getTeamsData.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/store/storemodels"
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+const (
+	// Query params
+	QueryParamPerPage = "per_page"
+	QueryParamPage    = "page"
+
+	// Pagination
+	DefaultPage         = 0
+	DefaultPerPageLimit = 10
+	MaxPerPageLimit     = 100
+)
+
+func (p *Plugin) GetMSTeamsTeamAndChannelDetailsFromChannelLinks(channelLinks []*storemodels.ChannelLink, userID string, checkChannelPermissions bool) (map[string]string, map[string]string, error) {
+	msTeamsTeamIDsVsNames := make(map[string]string)
+	msTeamsChannelIDsVsNames := make(map[string]string)
+	msTeamsTeamIDsVsChannelsQuery := make(map[string]string)
+
+	for _, link := range channelLinks {
+		if checkChannelPermissions && !p.API.HasPermissionToChannel(userID, link.MattermostChannelID, model.PermissionCreatePost) {
+			p.API.LogInfo("User does not have the permissions for the requested channel", "UserID", userID, "ChannelID", link.MattermostChannelID)
```

Comment:
do we really need this?

Reply:
Are you talking about the log statement of the check?